frontend/src/metabase/entities/users.js

```diff
@@ -89,11 +89,12 @@ const Users = createEntity({
         ...result,
       };
     },
-    update: thunkCreator => (...args) => async (dispatch, getState) => {
-      const result = await thunkCreator(...args)(dispatch, getState);
-      // HACK: reload memberships when updating a user
-      // TODO: only do this if group_ids changes
-      dispatch(loadMemberships());
+    update: thunkCreator => user => async (dispatch, getState) => {
+      const result = await thunkCreator(user)(dispatch, getState);
+      if (user.group_ids) {
```

Comment:
It'd be nice to actually compare any `group_ids` passed to this with those currently stored in state. However, we store the current user both nested in entities and at the top level. We can't count on either being present, so it'd be ugly logic to extract the current `group_ids`.

Instead we just check if `group_ids` is passed to update. I added a test that will fail if the normal user update form starts fetching memberships again.